lib/chef/resource_resolver.rb

```diff
@@ -84,9 +69,48 @@ def maybe_dynamic_resource_resolution(resource)
       handlers[0]
     end
 
-    # try the old static lookup of resources by mangling name to resource klass
-    def maybe_chef_platform_lookup(resource)
-      Chef::Resource.resource_matching_short_name(resource)
```

Comment:
So why not just throw a deprecation inside of `resource_matching_short_name()` instead of refactoring all of this code?  Then when the deprecation time is over, just remove this bit of code and remove resource_matching_short_name instead of trying to wire that up to the resource resolver.  I'm not sure what your refactoring is offering that wasn't here before and there might be edge conditions with the way that you're always adding the resource class now.